src/TraceEvent/EventPipe/EventPipeEventSource.cs

```diff
@@ -8,11 +10,15 @@
 using System.Diagnostics;
 using System.Runtime.InteropServices;
 
+using nint = System.Int32; // as of today the deserializer is limited to 32 bit, in case we change in the future we don't want to miss any cast
```

Comment:
Don't bother with nint.   While we might relax the serialization so that the TOTAL size of the file is > 4Gig (in that case StreamLabel needs to be bigger), nint today is only used to measure the size of single event, and I think we can safely assume we are not likely to make THAT bigger than 4Gig.

Reply:
you are right, it was over-engineering